### PR TITLE
ssl: Add UG entry about signatue algorithm configuration

### PR DESCRIPTION
**Pull request by IngelaAndin**
User guide addition about signature algorithms

### CODE REVIEW

File: lib/ssl/doc/src/using_ssl.xml [hunk 1]
```diff
@@ -247,7 +247,56 @@ ok</code>
...
+    that in TLS-1.2 the signature algorithms chosen by the server will
+    also be affected by the chiper suite that is chosen, which is not
+    the case in TLS-1.3.
+  </p></note>
+    
+    <p> Using the function <seemfa
+    marker="ssl:ssl#signature_algs/2"><c>ssl:signature_algs/2</c></seemfa>
+    will let you inspect diffrent aspects of possible configurations
+    for your system. For example if TLS-1.3 and TLS-1.2 is supported the
+    default signature_algorithm list in OTP-26 and cryptolib form
```
Comment (KennethL) [inline lib/ssl/doc/src/using_ssl.xml:275]: form -> from

File: lib/ssl/doc/src/using_ssl.xml [hunk 2]
```diff
@@ -247,7 +247,56 @@ ok</code>
...
+  rsa_pkcs1_sha512,rsa_pkcs1_sha384,rsa_pkcs1_sha256
+  %% TLS 1.2 algorithms
+  {sha512,ecdsa},
+  {sha384,ecdsa},
+  {sha256,ecdsa}]
+   </code>
+   
+    <p>If you want to add support for a non default supported
+    algorithm you should append it to the default list as the
+    configuration is in prefered order. See also  <seemfa
```
Comment (KennethL) [inline lib/ssl/doc/src/using_ssl.xml:296]:
```suggestion
    configuration is in prefered order. like this:</p>
    <code>
    MySignatureAlgs = ssl:signature_algs(default, 'tlsv1.3') ++ [rsa_pkcs1_sha512, rsa_pkcs1_sha384],
    ssl:connect(Ip,Port,[{signature_algs, MySignatureAlgs,...]}),
    ...
    </code>
    <p>See also  <seemfa
```

**Review by KennethL — CHANGES_REQUESTED** [review]

File: lib/ssl/doc/src/using_ssl.xml (hunk 1 above)
Comment (IngelaAndin) [reply lib/ssl/doc/src/using_ssl.xml:275]: Good catch, fixed

File: lib/ssl/doc/src/using_ssl.xml [hunk 3]
```diff
@@ -247,7 +247,64 @@ ok</code>
...
+    schemes in TLS-1.3) is something that can be negotiated and hence
+    also configured. These algorithms/schemes will be used for digital
+    signatures in protocol messages and in certificates. </p>
+
+    <note><p>TLS-1.3 schemes have atom names whereas TLS-1.2
+    configuration is two tuples composed by one hash algorithm and one
+    signature algorithm. When both versions are supported the
+    configuration can be a mix of these as both versions might be
+    negotiated. All <c>rsa_pss</c> based schemes are back ported to
+    TLS-1.2 and can be used aslo in a TLS-1.2 configuration. Also note
```
Comment (u3s) [inline lib/ssl/doc/src/using_ssl.xml:265]: Not an expert on grammar, but shouldn't you put a coma (`,`) after "Also note" ?

File: lib/ssl/doc/src/using_ssl.xml [hunk 4]
```diff
@@ -247,7 +247,64 @@ ok</code>
...
+    that in TLS-1.2 the signature algorithms chosen by the server will
+    also be affected by the chiper suite that is chosen, which is not
+    the case in TLS-1.3.
+  </p></note>
+    
+    <p> Using the function <seemfa
+    marker="ssl:ssl#signature_algs/2"><c>ssl:signature_algs/2</c></seemfa>
+    will let you inspect diffrent aspects of possible configurations
+    for your system. For example if TLS-1.3 and TLS-1.2 is supported the
```
Comment (u3s) [inline lib/ssl/doc/src/using_ssl.xml:274]: coma after "For example" ?

File: lib/ssl/doc/src/using_ssl.xml [hunk 5]
```diff
@@ -247,7 +247,64 @@ ok</code>
...
+  rsa_pkcs1_sha512,rsa_pkcs1_sha384,rsa_pkcs1_sha256
+  %% TLS 1.2 algorithms
+  {sha512,ecdsa},
+  {sha384,ecdsa},
+  {sha256,ecdsa}]
+   </code>
+   
+    <p>If you want to add support for non default supported
+    algorithms you should append them to the default list as the
+    configuration is in prefered order, like this:</p>
```
Comment (u3s) [inline lib/ssl/doc/src/using_ssl.xml:296]: typo: prefered -> preferred?

**Review by u3s — APPROVED** [review]

File: lib/ssl/doc/src/using_ssl.xml [hunk 6]
```diff
@@ -247,7 +247,64 @@ ok</code>
...
+    <section>
+    <title>Customizing signature algorithms(TLS-1.2)/schemes(TLS-1.3)</title>
+
+    <p> Starting from TLS-1.2 signature algorithms (called signature
+    schemes in TLS-1.3) is something that can be negotiated and hence
+    also configured. These algorithms/schemes will be used for digital
+    signatures in protocol messages and in certificates. </p>
+
+    <note><p>TLS-1.3 schemes have atom names whereas TLS-1.2
+    configuration is two tuples composed by one hash algorithm and one
```
[the remark this answers is not in the document]
Comment (IngelaAndin) [reply lib/ssl/doc/src/using_ssl.xml:261]: Yes that is what I mean, I guess leaving element out is a bit sloppy and would actually not have the intended meaning! I will add element to the sentence!

File: lib/ssl/doc/src/using_ssl.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (IngelaAndin) [reply lib/ssl/doc/src/using_ssl.xml:265]: fixed

File: lib/ssl/doc/src/using_ssl.xml (hunk 4 above)
Comment (IngelaAndin) [reply lib/ssl/doc/src/using_ssl.xml:274]: Reformulated a bit!

File: lib/ssl/doc/src/using_ssl.xml (hunk 5 above)
Comment (IngelaAndin) [reply lib/ssl/doc/src/using_ssl.xml:296]: fixed

**Review by dgud — APPROVED** [review]

**Pull request merged**